src/hmem_cuda.c

```diff
@@ -613,6 +636,92 @@ static int cuda_hmem_detect_p2p_access_support(void)
 	return FI_SUCCESS;
 }
 
+/**
+ * @brief detect dmabuf support in the current platform
+ * This checks the dmabuf support in the current platform
+ * by querying the property of cuda device 0
+ *
+ * @return  FI_SUCCESS if dmabuf support check is successful
+ *         -FI_EIO upon CUDA API error
+ */
+static int cuda_hmem_detect_dmabuf_support(void)
+{
+	CUresult cuda_ret;
+	CUdevice dev;
+	int is_supported = 0;
+
+	if (cuda_attr.device_count <= 1)
```

Comment:
Is it fair to not even check DMA buf support for single GPU instances? It makes sense for GDR but we may use DMA buf with EFA NIC for HPC applications.

Reply:
I think this line need to be removed. I was copying something from detect_p2p_support

Reply:
Removed